protocols/identify/src/behaviour.rs

```diff
@@ -242,6 +225,14 @@ impl Behaviour {
             }
         }
     }
+
+    fn all_addresses(&self) -> HashSet<Multiaddr> {
+        self.listen_addresses
+            .iter()
+            .chain(self.external_addresses.iter())
+            .cloned()
+            .collect()
```

Comment:
Just a note for sometime in the future, ideally we would prioritize the addresses we send to a remote, e.g. we should prioritize (in terms of order in the list) external addresses over listening addresses. Again, not for this pull request.

Reply:
That is an easy enough change!

Reply:
It actually isn't because we are collecting in a `HashSet` and thus lose any ordering.